src/runaqa.ts

```diff
@@ -1,324 +1,388 @@
+      .replace(/\r\n/g, '\n')
+      .split('\n')
+      .filter(Boolean)
+    for (const l of lines) {
+      const regexp = /TEST_JDK_HOME=(.*)/
+      const match = regexp.exec(l)
+      if (match && match[1]) {
+        process.env.TEST_JDK_HOME = match[1]
+        core.info(`Reset TEST_JDK_HOME to ${process.env.TEST_JDK_HOME}`)
+      }
+    }
+  }
+}
+
+function getTestJdkHome(version: string, jdksource: string): string {
+  // Try JAVA_HOME first and then fall back to GITHUB actions default location
+  let javaHome = process.env[`JAVA_HOME_${version}_X64`] as string
+  if (javaHome === undefined) {
+    javaHome = process.env['JAVA_HOME'] as string
+  }
+  if (jdksource === 'install-jdk') {
+    // work with AdoptOpenJDK/install-sdk
+    if (`JDK_${version}` in process.env) {
+      javaHome = process.env[`JDK_${version}`] as string
+    } else {
+      javaHome = process.env.JAVA_HOME as string
+    }
+  }
+  // Remove spaces in Windows path and replace with a short name path, e.g. 'C:/Program Files/***' ->C:/Progra~1/***
+  if (IS_WINDOWS && jdksource === 'github-hosted') {
+    javaHome = javaHome.replace(/Program Files/g, 'Progra~1')
+  }
+  if (javaHome === undefined) {
+    core.error('JDK could not be found')
+  }
+  return javaHome
+}
+
+// This function is an alternative of extra install step in workflow or alternative install action. This could also be implemented as github action
+async function installDependencyAndSetup(): Promise<void> {
+  if (IS_WINDOWS) {
+    const cygwinPath = 'C:\\cygwin64'
+    try {
+      if (!fs.existsSync(cygwinPath)) {
+        core.info(`if the cygwin exist?`)
+        await io.mkdirP('C:\\cygwin64')
+        await io.mkdirP('C:\\cygwin_packages')
+        await tc.downloadTool(
+          'https://cygwin.com/setup-x86_64.exe',
+          'C:\\temp\\cygwin.exe'
+        )
+        await exec.exec(`C:\\temp\\cygwin.exe  --packages wget,bsdtar,rsync,gnupg,git,autoconf,make,gcc-core,mingw64-x86_64-gcc-core,unzip,zip,cpio,curl,grep,perl --quiet-mode --download --local-install
+        --delete-orphans --site  https://mirrors.kernel.org/sourceware/cygwin/
+        --local-package-dir "C:\\cygwin_packages"
+        --root "C:\\cygwin64"`)
+        await exec.exec(
+          `C:/cygwin64/bin/git config --system core.autocrlf false`
+        )
+        core.addPath(`C:\\cygwin64\\bin`)
+      }
+    } catch (error) {
+      if (error instanceof Error){
+        core.warning(error.message)
+      }
+      else{
+        core.warning('Unexpected error')
+      }
+    }
+    const antContribFile = await tc.downloadTool( 
+      `https://sourceforge.net/projects/ant-contrib/files/ant-contrib/ant-contrib-1.0b2/ant-contrib-1.0b2-bin.zip/download`
+    )
+    await tc.extractZip(`${antContribFile}`, `${tempDirectory}`)
+    await io.cp(
+      `${tempDirectory}/ant-contrib/lib/ant-contrib.jar`,
+      `${process.env.ANT_HOME}\\lib`
+    )
+  } else if (process.platform === 'darwin') {

```

Comment:
Would be good to have all of the O/S derivates - so it's clear that it's isMac, isSolaris, isLinux, isWindows and so forth